### DISCUSSION TITLE
Can you make a notebook start by only clicking on a link?

### DISCUSSION THREAD

**Discussion by sloede** [fonsp/Pluto.jl#2418] [Questions about using Pluto] [answered]
I am using Pluto.jl mainly for teaching at university, specifically in a classroom setting where I prepare notebooks that I show in class to illustrate certain principles, and where I hope that students will also experiment themselves with the notebooks on their own laptops.

I know it is already possible to load a notebook by giving the URL to the students and then have them paste it into the Pluto landing page. However, I was wondering if this can be made even easier by giving them just a link that they can click and that will start up Julia, load Pluto, and then run Pluto with a notebook loaded?

This would make it even easier to share notebooks with an audience and further lower the entry barrier for new users.

**Comment by fonsp** [comment, marked as answer]
Hey @sloede ! This is not possible, as that would be a security problem 😢

Have you tried the integrated binder feature for HTML exports? https://github.com/JuliaPluto/static-export-template makes it easy to generate a site automatically, and all notebooks will contain embedded download and binder links. See https://www.youtube.com/watch?v=HiI4jgDyDhY&t=473s to learn more

**Reply by sloede** [reply]
Thanks, and yes, I am using the HTML exports. But why would starting it by clicking on a link a security problem? I am thinking about Zoom links, (which I believe start with `zoommtg://...`), which also prompt the user if you really want to start a given program. Wouldn't it be possible to do something similar for Julia and Pluto.jl?

**Reply by fonsp** [reply]
Exactly! We want to add a prompt like that, #259. Registering a URI handler like zoom is something we might add to our native app: https://github.com/JuliaPluto/PlutoDesktop

We do need to take security very seriously though, it is very easy to trick people into installing a virus by hiding/obfuscating bad code in a normal-looking notebook or package.

**Reply by sloede** [reply]
I understand the security concerns, and I think having a one-click startup (once either Julia or PlutoDesktop are properly installed) would greatly benefit the use of Pluto in - at least - the teaching realm.

**Reply by sloede** [reply]
Anyhow, just wanted to say that this is an awesome package and keep up the good work 👍
